File: src/ios/swift/AboutView.swift
import SwiftUI

struct AboutView: View {

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Image("icon")
                .resizable()
                .scaledToFit()
                .frame(maxWidth: .infinity)

            Text("Butterscotch iOS")
                .font(.largeTitle)
                .fontWeight(.bold)

            Text("commit \(BuildInfo.commitHash)")
                .font(.footnote)
                .fontWeight(.light)

            Divider()

            
            Text("A fork of Butterscotch runner, an open source re-implementation of GameMaker: Studio's runner (YoYo Runner), adding iOS support")

            HStack {
                Link("GitHub", destination: URL(string: "https://github.com/ChickenChunk579/Butterscotch-iOS")!)
                
                Link("Upstream", destination: URL(string: "https://github.com/ButterscotchRunner/Butterscotch")!)
            }
            .frame(maxWidth: .infinity)
        }
        .padding()
    }
}

File: android/app/src/main/kotlin/com/butterscotch/butterscotch/AboutView.kt
package com.butterscotch.butterscotch

import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

@Composable
fun AboutView(padding: PaddingValues) {
    val context = LocalContext.current

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(padding)
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(10.dp),
        horizontalAlignment = Alignment.Start
    ) {
        Image(
            painter = painterResource(R.drawable.icon),
            contentDescription = "Butterscotch icon",
            modifier = Modifier
                .fillMaxWidth()
                .weight(1f),
            contentScale = ContentScale.Fit
        )

        Text(
            text = "Butterscotch Android",
            style = MaterialTheme.typography.headlineLarge,
            fontWeight = FontWeight.Bold
        )

        Text(
            text = "commit unknown",
            style = MaterialTheme.typography.bodySmall,
            fontWeight = FontWeight.Light
        )

        HorizontalDivider()

        Text(
            text = "A fork of Butterscotch runner, an open source re-implementation of GameMaker: Studio's runner (YoYo Runner), adding Android support",
            style = MaterialTheme.typography.bodyLarge
        )

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.Center
        ) {
            TextButton(
                onClick = {
                    context.startActivity(
                        Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse("https://github.com/ChickenChunk579/Butterscotch-iOS")
                        )
                    )
                }
            ) {
                Text("GitHub")
            }

            TextButton(
                onClick = {
                    context.startActivity(
                        Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse("https://github.com/ButterscotchRunner/Butterscotch")
                        )
                    )
                }
            ) {
                Text("Upstream")
            }
        }
    }
}